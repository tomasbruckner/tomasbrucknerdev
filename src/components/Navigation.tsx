import React, { FC, useState } from 'react';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/ViewHeadline';
import IconButton from '@material-ui/core/IconButton';
import { makeStyles, useMediaQuery, useTheme } from '@material-ui/core';
import { useCommonStyles } from '../styles/commonStyles';
import { t } from '../utils/i18n';

const useStyles = makeStyles((theme) => ({
  menuIcon: {
    fill: theme.palette.common.white,
  },
}));

const Navigation: FC = () => {
  const [anchor, setAnchor] = useState<HTMLElement | null>(null);
  const theme = useTheme();
  const isLg = useMediaQuery(theme.breakpoints.up('lg'));
  const classes = useStyles();
  const commonClasses = useCommonStyles();

  if (isLg) {
    return (
      <>
        <a className={commonClasses.menuLinkStyle} href="#about">
          {t('navigation.about')}
        </a>
        <a className={commonClasses.menuLinkStyle} href="#contact">
          {t('navigation.contact')}
        </a>
        <a className={commonClasses.menuLinkStyle} href="#lectures">
          {t('navigation.lectures')}
        </a>
      </>
    );
  }

  return (
    <>
      <IconButton
        onClick={(event: React.MouseEvent<HTMLElement>) => setAnchor(event.currentTarget)}
      >
        <MenuIcon className={classes.menuIcon} />
      </IconButton>
      <Menu
        open={!!anchor}
        onClose={() => setAnchor(null)}
        anchorEl={anchor}
        getContentAnchorEl={null}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'center',
        }}
      >
        <MenuItem key="about" component="a" href="#about" button onClick={() => setAnchor(null)}>
          {t('navigation.about')}
        </MenuItem>
        <MenuItem
          key="contact"
          component="a"
          href="#contact"
          button
          onClick={() => setAnchor(null)}
        >
          {t('navigation.contact')}
        </MenuItem>
        <MenuItem
          key="lectures"
          component="a"
          href="#lectures"
          button
          onClick={() => setAnchor(null)}
        >
          {t('navigation.lectures')}
        </MenuItem>
      </Menu>
    </>
  );
};

export default Navigation;
