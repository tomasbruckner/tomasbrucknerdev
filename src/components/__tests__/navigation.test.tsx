jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (text: string) => text,
  }),
}));

import React from 'react';
import Navigation from 'components/Navigation';
import renderer from 'react-test-renderer';

it('Navigation renders correctly', () => {
  const tree = renderer.create(<Navigation />).toJSON();
  expect(tree).toMatchSnapshot();
});
