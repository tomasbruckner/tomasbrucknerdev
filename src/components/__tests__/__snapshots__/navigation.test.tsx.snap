// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
Array [
  <a
    href="#about"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": 500,
        "padding": 16,
        "textDecoration": "none",
      }
    }
  >
    navigation.about
  </a>,
  <a
    href="#contact"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": 500,
        "padding": 16,
        "textDecoration": "none",
      }
    }
  >
    navigation.contact
  </a>,
  <a
    href="#lectures"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": 500,
        "padding": 16,
        "textDecoration": "none",
      }
    }
  >
    navigation.lectures
  </a>,
]
`;
